lib/Conversion/HWToLLVM/HWToLLVM.cpp

```diff
@@ -342,18 +352,59 @@ struct HWConstantOpConversion : public ConvertToLLVMPattern {
 } // namespace
 
 namespace {
+
+static Namespace globals;
+
 /// Convert an ArrayOp operation to the LLVM dialect. An equivalent and
 /// initialized llvm dialect array type is generated.
 struct HWArrayCreateOpConversion
     : public ConvertOpToLLVMPattern<hw::ArrayCreateOp> {
   using ConvertOpToLLVMPattern<hw::ArrayCreateOp>::ConvertOpToLLVMPattern;
 
-  LogicalResult
-  matchAndRewrite(hw::ArrayCreateOp op, OpAdaptor adaptor,
-                  ConversionPatternRewriter &rewriter) const override {
+  bool isConst(hw::ArrayCreateOp op) const {
+    for (auto op : op.getOperands()) {
+      if (!op.getDefiningOp() || !isa<hw::ConstantOp>(op.getDefiningOp())) {
+        return false;
+      }
+    }
+    return true;
+  }
 
-    auto arrayTy = typeConverter->convertType(op->getResult(0).getType());
+  Value convertConstArray(hw::ArrayCreateOp &op, OpAdaptor &adaptor,
+                          ConversionPatternRewriter &rewriter,
+                          Type arrayTy) const {
+    OpBuilder b(op->getParentOfType<mlir::ModuleOp>().getBodyRegion());
+
+    auto name = globals.newName(Twine("array_global"));
```

Comment:
Nit: Isn't there an implicit conversion such that we can just write "array_global"?

Reply:
Hmm I'm not sure what you mean. If I just use "array_global" without the namespace there are errors about symbols being defined multiple times. Is there supposed to be something that makes global names unique automatically?

Reply:
Sorry for being imprecise, I thought about the following change:
```suggestion
    auto name = globals.newName("array_global");
```